### ISSUE TITLE
Bug: [ban-types] false positive caused by `Object` as a generic type name

### ISSUE THREAD

**Issue by alex-sandri** [typescript-eslint/typescript-eslint#5505]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.4&sourceType=module&code=C4TwDgpgBACghgJzgWwM4B4DCALAlgGwBMAaKAeQCMArCAY2AD4oBeKAbwFgAoKKWvIgC4oOAoQDc3XgHtqdYMMo16krgF9VQA&eslintrc=N4KABGBECmAeAu0B2ATAzpAXGA2uCU0aANgJZLyYBO0AxgPYC2jyK0KkANPhJAA7EArgHNymAALwAnnyK0qpPvAC0RMhQD0NBs1btI+ALrdefAIZU00KliiSZchUtUly8DecvWu+fkNFIGNh4BHbSsmjyiipqbgYQhiAAvkA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkeAhpgObp4VTokBMADAIwA2SABpwNSPFgATAK4Z2MBJKIArMqJocSAWWlz0ASSJ4ssROzyYZ6DeIBmsTNHQBhYiWQkTxl-Q9EmIwAxNHQAOXp4TgsrGzFaL0xkaDZKS2tbBIBrZEQAGWQAIxcAC3RoLJjrMQBfEBqgA

### Repro Code

```typescript
type Params<Child, Object> = {
  child: Child;
  object: Object;
};
```

### ESLint Config

```javascript
module.exports = {
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "plugins": [
        "@typescript-eslint"
    ],
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "target": "es2016",
    "module": "commonjs",
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "skipLibCheck": true
  }
}
```

### Expected Result

I expected no error.

For instance this code works:

```typescript
type Params<T1, T2> = {
  child: T1;
  object: T2;
};

```

### Actual Result

3:11  error    Don't use `Object` as a type. The `Object` type actually means "any non-nullish value", so it is marginally better than `unknown`.
- If you want a type meaning "any object", you probably want `Record<string, unknown>` instead.
- If you want a type meaning "any value", you probably want `unknown` instead  @typescript-eslint/ban-types

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.33.1` |
| `@typescript-eslint/parser`            | `5.33.1` |
| `TypeScript`                           | `4.7.4` |
| `ESLint`                               | `8.22.0` |
| `node`                                 | `16.14.0` |

**Comment by bradzacher** [comment]
You should avoid shadowing common global names - it's a code smell that can really create confusing code because users need to trace the names to see if they're global or not.

---

This is working as intended though - the rule simply checks and bans type names. The context in which they're defined is irrelevant. This makes it possible to ban names from being used at all in the codebase in addition to banning global types.
